### PR TITLE
EVAKA-4103 application attachments

### PR DESCRIPTION
**Pull request by hruoho**
#### Summary

Add component for including attachments. Also tweak service need according to figma specs.

Screenshot 2021-01-27 at 13 48 09

### CODE REVIEW

File: frontend/packages/citizen-frontend/src/applications/editor/ApplicationFormData.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import {
+  Attachment,
   Application,
   ApplicationForm,
   ApplicationPersonBasics
 } from '~applications/types'
+import { UUID } from '~../../lib-common/src/types'
 
 export type ServiceNeedFormData = {
+  applicationId: UUID
```
Comment (Joosakur) [inline frontend/packages/citizen-frontend/src/applications/editor/ApplicationFormData.ts:11]: I don't think applicationId should be part of form data. Rather pass it in props or read it from router.

File: frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        const others = old.filter((item) => item.id !== fileObject.id)
+        const progressedFile = { ...fileObject, error: 'server-error' }
+        return [...others, progressedFile]
+      })
+    }
+  }
+
+  return (
+    <FileUploadContainer>
+      <FileInputLabel className="file-input-label" onDrop={onDrop}>
```
Comment (Joosakur) [inline frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx:289]: `input` and `h4` inside `label` sounds kinda exotic. Not sure if that may cause accessibility issues or such.

Comment (hruoho) [reply frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx:289]: I think `input` is quite often seen inside `label` but this `h4` is truly a funky one.

**Comment by Joosakur** [comment]
bug: 

1. add attachment
2. remove the attachment
3. try to add the **same** file again --> nothing happens

**Comment by Joosakur** [comment]
bug:

1. set network to offline
2. add attachment --> request fails silently
3. set network to online and try again --> nothing happens

**Comment by Joosakur** [comment]
ux question for future consideration: if you uncheck urgent care and save the application, should the related attachments be deleted? user no longer sees them so he/she likely expects that they are no longer included

File: frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+  cursor: pointer;
+
+  &:hover {
+    color: ${colors.blues.medium};
+  }
+
+  &:disabled {
+    color: ${colors.greyscale.lighter};
+    cursor: not-allowed;
+  }
```
Comment (Joosakur) [inline frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx:149]: Just giving it IconButton `grey=true` property makes these unnecessary I think

File: frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+  flex-direction: column;
+  flex-wrap: nowrap;
+`
+
+const FileHeader = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  justify-content: space-between;
+  font-size: 15px;
```
Comment (Joosakur) [inline frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx:103]: I think File, FileDetails and/or FileHeader might need `align-items: center;` to have everything vertically aligned.

File: frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx (hunk 3 above)
Comment (hruoho) [reply frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx:149]: Almost... This seems to force gray on hovered element too so I'll skip this now.

**Review by Joosakur — APPROVED** [review]
🏅

**Pull request merged**